File: front/src/app/core/models/solicitacao-projeto.ts
import { PerfilEnum } from '../enums';

export class SolicitacaoProjeto {
  public id?: number;
  public tituloProjeto!: string;
  public descricaoProjeto!: string;
  public responsavelProjeto?: string;
  public motivacaoProjeto?: string;
  public objetivosEspecificosProjeto!: string;
  public objetivosEstrategicosSesa!: string;
  public premissas?: string;
  public riscos?: string;
  public restricoes?: string;
  public aprovacao?: number;
  public projetoViavel?: number;
  public estimativaTempo?: string;
  public estimativaCusto?: string;
  public ativo?: string;

  constructor(init?: Partial<SolicitacaoProjeto>) {
    Object.assign(this, init);
  }
}

File: back/src/main/java/br/gov/ce/saude/tsp/domains/entities/SolicitacaoProjeto.java
package br.gov.ce.saude.tsp.domains.entities;

import br.gov.ce.saude.tsp.common.Auditoria;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.util.List;

@Setter
@Getter
@Entity
@Table(name = "solicitacao_projeto")
public class SolicitacaoProjeto extends Auditoria {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "titulo_projeto", nullable = false)
    private String tituloProjeto;

    @Column(name = "descricao_projeto", nullable = false)
    private String descricaoProjeto;

    @Column(name = "responsavel_projeto", nullable = false)
    private String responsavelProjeto;

    @Column(name = "motivacao_projeto")
    private String motivacaoProjeto;

    @Column(name = "objetivos_especificos_projeto", length = 1200)
    private String objetivosEspecificosProjeto;

    @ManyToMany
    @JoinTable(
            name = "solicitacao_objetivo_estrategico",
            joinColumns = @JoinColumn(name = "solicitacao_id"),
            inverseJoinColumns = @JoinColumn(name = "objetivo_estrategico_id"),
            uniqueConstraints = @UniqueConstraint(columnNames = {"solicitacao_id", "objetivo_estrategico_id"})
    )
    private List<ObjetivosEstrategicos> objetivosEstrategicosSesa;

    @Column(name = "premissas", length = 1200)
    private String premissas;

    @Column(name = "riscos")
    private String riscos;

    @Column(name = "restricoes")
    private String restricoes;

    @Column(name = "aprovacao")
    private Boolean aprovacao;

    @Column(name = "projeto_viavel")
    private Boolean projetoViavel;

    @Column(name = "estimativa_tempo")
    private String estimativaTempo;

    @Column(name = "estimativa_custo")
    private String estimativaCusto;

    @Column(nullable = false)
    private Boolean ativo = true;
}